File: src/main/java/hex/paymentstore/domain/AddPaymentUseCase.java
package hex.paymentstore.domain;

import org.apache.commons.lang3.RandomStringUtils;

public class AddPaymentUseCase {
	private final PaymentRepository repository;
	
	public AddPaymentUseCase(PaymentRepository repository) {
		this.repository = repository;
	}
	
	public Payment add(Payment payment) {
		String id = RandomStringUtils.randomAlphanumeric(20);
		return repository.add(payment.copy(id));
	}
}

File: src/main/kotlin/hex/paymentstore/domain/AddPaymentUseCase.kt
package hex.paymentstore.domain

import org.apache.commons.lang3.RandomStringUtils

class AddPaymentUseCase(private val repository: PaymentRepository) {
	fun add(payment: Payment): Payment {
		val id: String = RandomStringUtils.randomAlphanumeric(20)
		return repository.add(payment.copy(id = id))
	}
}